### PR TITLE
Refactor: move dask utils from misc.py to new dask_utils.py

### PR DESCRIPTION
**Pull request by pwinston**
# Description
Move 4 dask cache related functions from `misc.py` to new `dask.py`.

## Type of change
- [x] Refactor

# How has this been tested?
- [x] ran by hand
- [x] ran existing tests

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
Great to pull this code out into its own file.

@jni is there concern about potential confusion / mistakes by having a file called `dask.py`? (i.e. `from .dask import` now looks very similar to `from dask import`). We could name the file `dask_extras.py` or `dask_utils.py` for example just to be really clear, but maybe that is a bit verbose for something already in a `utils` directory. Otherwise I think this is good to go.

File: napari/utils/dask.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+def _is_dask_data(data) -> bool:
+    """Return True if data is a dask array or a list/tuple of dask arrays."""
+    return isinstance(data, da.Array) or (
+        isinstance(data, (list, tuple))
+        and any(isinstance(i, da.Array) for i in data)
+    )
+
+
+def configure_dask(data) -> ContextManager[dict]:
```
Comment (tlambert03) [inline napari/utils/dask.py]:
```suggestion
def configure_dask(data) -> Callable[[], ContextManager[dict]]:
```
mind fixing this for me while you're at it?

File: napari/utils/dask.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                f'{dask.__version__}'
+            )
+
+        def dask_optimized_slicing(*args, **kwds):
+            with dask.config.set({"optimization.fuse.active": False}) as cfg:
+                yield cfg
+
+    else:
+
+        def dask_optimized_slicing(*args, **kwds):
```
Comment (tlambert03) [inline napari/utils/dask.py]:
```suggestion
        def dask_optimized_slicing():
```

File: napari/utils/dask.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        create_dask_cache()  # creates one if it doesn't exist
+        dask_version = tuple(map(int, dask.__version__.split(".")))
+        if dask_version < (2, 15, 0):
+            warnings.warn(
+                'For best performance with Dask arrays in napari, please '
+                'upgrade Dask to v2.15.0 or later. Current version is '
+                f'{dask.__version__}'
+            )
+
+        def dask_optimized_slicing(*args, **kwds):
```
Comment (tlambert03) [inline napari/utils/dask.py]:
```suggestion
        def dask_optimized_slicing():
```

File: napari/utils/dask.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
+"""Dask cache utilities.
+"""
+from contextlib import contextmanager
+from typing import ContextManager, Optional
```
Comment (tlambert03) [inline napari/utils/dask.py]:
```suggestion
from typing import Callable, ContextManager, Optional
```

**Review by tlambert03 — APPROVED** [review]
looks good to me.  (Also fine with `dask_utils.py`)
I've been meaning to change the return type on that `dask_optimized_slicing` function... mind incorporating that here?

**Comment by pwinston** [comment]
Renamed to dask_utils.py and took @tlambert03's changes.

File: napari/utils/dask_utils.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        created, but not caching will occur. by default, cache size is
+        autodetermined using ``mem_fraction``.
+    mem_fraction : float, optional
+        The fraction (from 0 to 1) of total memory to use for the dask cache.
+        by default, 50% of total memory is used.
+
+    Returns
+    -------
+    [type]
+        [description]
```
Comment (jni) [inline napari/utils/dask_utils.py:32]: ??? :grimacing:

Comment (jni) [reply napari/utils/dask_utils.py:32]: Just saw that this existed before. Happy to leave a fix to a later PR.

**Review by jni — APPROVED** [review]

**Pull request merged**